File: backend/app/models/rating.py
from datetime import datetime

from sqlalchemy import DateTime, ForeignKey, Integer, UniqueConstraint
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.db.base import Base


class Rating(Base):
    __tablename__ = "ratings"

    id: Mapped[int] = mapped_column(primary_key=True, index=True)
    profile_id: Mapped[int] = mapped_column(ForeignKey("profiles.id"), nullable=False)
    content_id: Mapped[int] = mapped_column(ForeignKey("content.id"), nullable=False)
    score: Mapped[int] = mapped_column(Integer, nullable=False)
    created_at: Mapped[datetime] = mapped_column(DateTime, default=datetime.utcnow)

    profile = relationship("Profile", back_populates="ratings")
    content = relationship("Content", back_populates="ratings")

    __table_args__ = (
        UniqueConstraint("profile_id", "content_id", name="uq_profile_content_rating"),
    )

File: frontend/src/types/rating.ts
import type { Content } from "@/types/content";

export type Rating = {
  id: number;
  profile_id: number;
  content_id: number;
  score: number;
  created_at: string;
  content: Content | null;
};

export type RatingCreateInput = {
  profile_id: number;
  content_id: number;
  score: number;
};

export type RatingUpdateInput = {
  score: number;
};